src/engine/platform/es5506.cpp

```diff
@@ -0,0 +1,845 @@
+  for (size_t h=start; h<start+len; h++) {
+    // convert 32 bit access to 8 bit host interface
+    while (!hostIntf32.empty()) {
+      QueuedHostIntf w=hostIntf32.front();
+      if (w.isRead && (w.read!=NULL)) {
+        hostIntf8.emplace(0,w.addr,w.read,w.mask);
+        hostIntf8.emplace(1,w.addr,w.read,w.mask);
+        hostIntf8.emplace(2,w.addr,w.read,w.mask);
+        hostIntf8.emplace(3,w.addr,w.read,w.mask,w.delay);
+      } else {
+        hostIntf8.emplace(0,w.addr,w.val,w.mask);
+        hostIntf8.emplace(1,w.addr,w.val,w.mask);
+        hostIntf8.emplace(2,w.addr,w.val,w.mask);
+        hostIntf8.emplace(3,w.addr,w.val,w.mask,w.delay);
+      }
+      hostIntf32.pop();
+    }
+    es5506.tick_perf();
+    bufL[h]=es5506.lout(0);
+    bufR[h]=es5506.rout(0);
+  }
+}
+
+void DivPlatformES5506::e(bool state)
+{
+  if (es5506.e_rising_edge()) {
+    if (cycle) { // wait until delay
+      cycle--;
+    } else if (!hostIntf8.empty()) {
+      QueuedHostIntf w=hostIntf8.front();
+      unsigned char shift=24-(w.step<<3);
+      if (w.isRead) {
+        *w.read=((*w.read)&(~((0xff<<shift)&w.mask)))|((es5506.host_r((w.addr<<2)+w.step)<<shift)&w.mask);
+        if (w.step==3) {
+          if (w.delay>0) {
+            cycle+=w.delay;
+          }
+          isReaded=true;
+        } else {
+          isReaded=false;
+        }
+        hostIntf8.pop();
+      } else {
+        isReaded=false;
+        unsigned int mask=(w.mask>>shift)&0xff;
+        if ((mask==0xff) || isMasked) {
+          if (mask==0xff) {
+            maskedVal=(w.val>>shift)&0xff;
+          }
+          es5506.host_w((w.addr<<2)+w.step,maskedVal);
+          if(dumpWrites) {
+            addWrite((w.addr<<2)+w.step,maskedVal);
+          }
+          isMasked=false;
+          if ((w.step==3) && (w.delay>0)) {
+            cycle+=w.delay;
+          }
+          hostIntf8.pop();
+        } else if (!isMasked) {
+          maskedVal=((w.val>>shift)&mask)|(es5506.host_r((w.addr<<2)+w.step)&~mask);
+          isMasked=true;
+        }
+      }
+    }
+  }
+  if (isReaded) {
+    isReaded=false;
+    if (irqTrigger) {
+      irqTrigger=false;
+      if ((irqv&0x80)==0) {
+        unsigned char ch=irqv&0x1f;
+        if (chan[ch].isReverseLoop) { // Reversed loop
+          pageWriteMask(0x00|ch,0x5f,0x00,(chan[ch].pcm.reversed?0x0040:0x0000)|0x08,0x78);
+          chan[ch].isReverseLoop=false;
+        }
+      }
+    }
+  }
+}
+
+void DivPlatformES5506::irqb(bool state) {
+  rRead(0x0e,&irqv,0x9f);
+  irqTrigger=true;
+}
+
+void DivPlatformES5506::tick(bool sysTick) {
+  for (int i=0; i<=chanMax; i++) {
+    chan[i].std.next();
+    DivInstrument* ins=parent->getIns(chan[i].ins);
+    // volume/panning macros
+    if (chan[i].std.vol.had) {
+      chan[i].outVol=((chan[i].vol&0xff)*MIN(0xffff,chan[i].std.vol.val))/0xff;
+      if (!isMuted[i]) {
+        chan[i].volChanged=true;
+      }
+    }
+    if (chan[i].std.panL.had) {
```

Comment:
You might want to unlock the full 65535-step panning range.

Reply:
Same as volume macro, It's now using log volume and unsigned int variables.